.github/workflows/benchmark-pr.yaml

```diff
@@ -0,0 +1,63 @@
+---
+name: Benchmarks on AMD64
+permissions: read-all
+on: [pull_request]
+jobs:
+  benchmark-pull-request:
+    runs-on: ubuntu-latest-8-cores
+    steps:
+    - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      with:
+        fetch-depth: 0
+    - id: goversion
+      run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+    - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: ${{ steps.goversion.outputs.goversion }}
+    - name: Install Benchstat
+      run: make install-benchstat
+    - name: Run Benchmarks
+      run: |
+        BENCHSTAT_FORMAT=csv BENCHSTAT_OUTPUT_FILE=result.csv make test-benchmark-compare REF=${{ github.event.pull_request.head.sha }}
+        echo "$(head -n1 result.csv),,," > output.csv # Add three missing cols from header
```

Comment:
I do not get time to try `benchstat`. We have already set `-format=csv` when running `benchstat`, I am curious why we need to add three missing cols from header. Isn't it a bug of `benchstat`?

Reply:
It's a combination of bugs which make it not being displayed friendly as a Markdown table:

1. `benchstat` omits three columns (see the attached [`output.csv`](https://github.com/etcd-io/bbolt/files/15476940/output.csv)) — this is technically still a valid CSV, as it the missing columns are part of a header row
2. The `petems/csv-to-md-table-action` creates the table with the missing columns in the first row (header, refer to https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables)
3. Then, GitHub doesn't display the Markdown table because the header doesn't match the maximum number of columns in the file.

I know this is a fragile part of the script. We could alternatively skip the Markdown table and display it with `benchstat` native text formatting:

```
                                 │     bm1      │                  bm2                  │
                                 │    sec/op    │    sec/op      vs base                │
_FreelistRelease10K-8              298.1µ ±  0%    298.9µ ±  0%       ~ (p=0.400 n=3) ¹
_FreelistRelease100K-8             4.081m ±  0%    4.099m ±  1%       ~ (p=1.000 n=3) ¹
_FreelistRelease1000K-8            56.74m ±  2%    66.39m ±  3%       ~ (p=0.100 n=3) ¹
_FreelistRelease10000K-8           921.7m ± 17%   1045.5m ±  6%       ~ (p=0.400 n=3) ¹
Bucket_CreateBucketIfNotExists-8   226.2µ ± 10%    245.6µ ± 13%       ~ (p=1.000 n=3) ¹
DBBatchAutomatic-8                 1.662m ±  1%    1.672m ±  1%       ~ (p=0.400 n=3) ¹
DBBatchSingle-8                    305.6m ±  4%    284.2m ±  8%       ~ (p=0.400 n=3) ¹
DBBatchManual10x100-8              4.929m ±  6%    4.701m ±  1%       ~ (p=0.700 n=3) ¹
geomean                            8.802m          9.091m        +3.29%
¹ need >= 4 samples to detect a difference at alpha level 0.05

                                 │     bm1      │                 bm2                  │
                                 │     B/op     │     B/op      vs base                │
_FreelistRelease10K-8              259.2Ki ± 0%   259.2Ki ± 0%       ~ (p=1.000 n=3) ¹
_FreelistRelease100K-8             2.295Mi ± 0%   2.295Mi ± 0%       ~ (p=0.200 n=3) ¹
_FreelistRelease1000K-8            29.07Mi ± 0%   29.07Mi ± 0%       ~ (p=0.200 n=3) ¹
_FreelistRelease10000K-8           251.0Mi ± 0%   251.0Mi ± 0%       ~ (p=1.000 n=3) ²
Bucket_CreateBucketIfNotExists-8   5.523Ki ± 0%   5.523Ki ± 0%       ~ (p=1.000 n=3) ²
DBBatchAutomatic-8                 424.0Ki ± 0%   424.0Ki ± 0%       ~ (p=0.700 n=3) ¹
DBBatchSingle-8                    18.03Mi ± 0%   18.11Mi ± 0%       ~ (p=0.100 n=3) ¹
DBBatchManual10x100-8              867.3Ki ± 2%   867.1Ki ± 2%       ~ (p=0.700 n=3) ¹
geomean                            1.862Mi        1.863Mi       +0.05%
¹ need >= 4 samples to detect a difference at alpha level 0.05
² all samples are equal

                                 │     bm1     │                 bm2                 │
                                 │  allocs/op  │  allocs/op   vs base                │
_FreelistRelease10K-8               30.00 ± 0%    30.00 ± 0%       ~ (p=1.000 n=3) ¹
_FreelistRelease100K-8             1.724k ± 0%   1.722k ± 0%       ~ (p=0.300 n=3) ²
_FreelistRelease1000K-8             38.00 ± 0%    38.00 ± 0%       ~ (p=1.000 n=3) ¹
_FreelistRelease10000K-8            50.00 ± 0%    50.00 ± 0%       ~ (p=1.000 n=3) ¹
Bucket_CreateBucketIfNotExists-8    41.00 ± 0%    41.00 ± 0%       ~ (p=1.000 n=3) ¹
DBBatchAutomatic-8                 10.17k ± 0%   10.17k ± 0%       ~ (p=0.500 n=3) ²
DBBatchSingle-8                    56.65k ± 0%   56.64k ± 0%       ~ (p=0.700 n=3) ²
DBBatchManual10x100-8              5.970k ± 1%   5.940k ± 2%       ~ (p=1.000 n=3) ²
geomean                             585.8         585.3       -0.08%
¹ all samples are equal
² need >= 4 samples to detect a difference at alpha level 0.05
```


Let me know your thoughts.

Reply:
> 1. this is technically still a valid CSV, as it the missing columns are part of a header row

Based on [rfc4180](https://www.ietf.org/rfc/rfc4180.txt),  the header line should contain the same number of fields as the records. Let's add a TODO, and raise an issue to benchstat, and see what's the response?


>  I know this is a fragile part of the script. We could alternatively skip the Markdown table and display it with benchstat native text formatting

I like the CSV/table output, which is clearer.

Reply:
> I like the CSV/table output, which is clearer.

But If we use the TEXT output, it can simplify the workflow. Will it be more difficult to validate results?

Reply:
What if I raise a bug for `benchstat` and use the TEXT output in the meantime? As @tjungblu, we would ideally lock `benchstat`'s version because of the CSV. Which may not be ideal.

Reply:
Opened issue golang/go#67760.

Reply:
Done, switched to TEXT output in the meantime.